woocommerce/woocommerce-ios
# Order Creation: Add bottom sheet to choose new order flow

Closes https://github.com/woocommerce/woocommerce-ios/issues/5404.

## Description

This PR adds logic to choose between Simple Payments and Create Order flows. It will show bottom sheet with both options only when both are available and enabled in experimental features.
When only one type is enabled - it will be launched straight away, without presenting bottom sheet.
When nothing is enabled - "plus" button will be hidden.

## How

Most logic is added to `AddOrderCoordinator`. Some additional conditions added in `OrdersTabbedViewController` to show correct accessibility hint for "plus" button.

`BottomSheetListSelectorViewController` updated to hide empty header if there is no title.

## Test

1. Launch app with all feature toggles switched off.
2. Verify that there is no plus button on orders list.
3. Enable only "Simple Payments".
4. Check that plus button appeared and opens "Simple Payments" feature.
5. Enable only "Order Creation".
6. Check that plus button appeared and opens New Order view.
7. Enable both "Simple Payments" and "Order Creation" in experimental features.
8. Confirm that tapping on "plus" button on orders screen shows flow picker (in bottom sheet).
9. Confirm that both option work when selected from bottom sheet.
10. Disable all experimental features.
11. Verify that there is no plus button on orders list. 
12. Open create products type sheet and verify that header with title is shown.

## Screenshots

iPhone|iPad
--|--
![Simulator Screen Shot - iPhone 13](https://user-images.githubusercontent.com/3132438/142429005-175ed4d8-03e4-442a-84e9-f59f17923055.png)|![Simulator Screen Shot - iPad Pro (11-inch) (3rd generation)](https://user-images.githubusercontent.com/3132438/142432154-7de479ab-69db-42ce-8fc6-2a6bf22e9a50.png)

---
